theano/tensor/elemwise_cgen.py

```diff
@@ -100,13 +100,13 @@ def make_checks(loop_orders, dtypes, sub):
             check += """
             if (%%(lv%(j0)s)s_n%(x0)s != %%(lv%(j)s)s_n%(x)s)
             {
-                PyErr_Format(PyExc_ValueError, "Input dimension mis-match. (input[%%%%i].shape[%%%%i] = %%%%i, input[%%%%i].shape[%%%%i] = %%%%i)",
+                PyErr_Format(PyExc_ValueError, "Input dimension mis-match. (input[%%%%i].shape[%%%%i] = %%%%li, input[%%%%i].shape[%%%%i] = %%%%li)",
                    %(j0)s,
                    %(x0)s,
-                   %%(lv%(j0)s)s_n%(x0)s,
+                   (long int) %%(lv%(j0)s)s_n%(x0)s,
```

Comment:
On 64-bit Windows, `npy_intp` is 64-bit, but `long int` is 32-bit, so @abergeron is correct.